src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -35,15 +35,25 @@ private async Task AnalyzeDocumentForKindAsync(TextDocument document, AnalysisKi
         {
             try
             {
-                if (!AnalysisEnabled(document, _documentTrackingService))
+                if (!document.SupportsDiagnostics())
+                {
+                    return;
+                }
+
+                var isActiveDocument = _documentTrackingService.TryGetActiveDocument() == document.Id;
+                var isOpenDocument = document.IsOpen();
```

Comment:
note: it's unclear how this actually works in practice.  i thought we didn't track this information across OOP (where the incremental analyzer is running).  is that not the case? does this actually work?

Reply:
Incremental analyzers still run in-proc. If and when it is moved to OOP, diagnostic incremental analyzer would certainly need the `IsOpen` value to be marshalled to OOP to ensure we don't execute analyzers on closed files.